internal/resolver/dns/dns_resolver_test.go

```diff
@@ -756,11 +757,11 @@ func TestDNSResolverExponentialBackoff(t *testing.T) {
 	defer func(nt func(d time.Duration) *time.Timer) {
 		newTimer = nt
 	}(newTimer)
-	timerChan := make(chan *time.Timer, 1)
+	timerChan := testutils.NewChannel()
 	newTimer = func(d time.Duration) *time.Timer {
 		// Will never fire on its own, allows this test to call timer immediately.
 		t := time.NewTimer(time.Hour)
-		timerChan <- t
+		timerChan.SendOrFail(t)
```

Comment:
This is probably not what you want.  SendOrFail will just not push to the channel if it is already full.  You might have meant `SendContext` and a `t.Error`?  But, it's probably fine to just block indefinitely here (`Send`).  The whole point is that your test shouldn't hang for the default timeout of 10 minutes when something goes wrong.

Reply:
Are you saying that what I have now is wrong? I.e. right now I have a 10 minute wait using a context that I pass to every receive. Are you saying to use something like defaultTestShortTimeout for both the sends and receives?